Composer/packages/client/src/components/Adapters/TeamsManifestGenerator.tsx

```diff
@@ -0,0 +1,131 @@
+  },
+  rootHovered: {
+    backgroundColor: 'unset',
+  },
+  rootPressed: {
+    backgroundColor: 'unset',
+  },
+};
+
+type TeamsManifestGeneratorProps = {
+  hidden: boolean;
+  botAppId?: string;
+  botDisplayName?: string;
+  onDismiss: () => void;
+};
+
+export const TeamsManifestGenerator = (props: TeamsManifestGeneratorProps) => {
+  const textFieldRef = useRef<ITextField>(null);
+
+  const copyCodeToClipboard = () => {
+    try {
+      textFieldRef.current?.select;
+      textFieldRef.current?.select();
+      document.execCommand('copy');
+      textFieldRef.current?.setSelectionRange(0, 0);
+      textFieldRef.current?.blur();
+    } catch (e) {
+      console.error('Something went wrong when trying to copy manifest content to clipboard.', e);
+    }
+  };
+
+  const generateTeamsManifest = () => {
+    const appId = props.botAppId ? props.botAppId : '{AddBotAppId}';
+    const botName = props.botDisplayName ? props.botDisplayName : '{AddBotDisplayName}';
+    const result = defaultTeamsManifest;
+    result.id = uuidv4().toString();
+    result.description.short = `short description for ${botName}`;
+    result.description.full = `full description for ${botName}`;
+    result.packageName = botName;
+    result.name.short = botName;
+    result.name.full = botName;
+    result.bots[0].botId = appId;
+    return JSON.stringify(defaultTeamsManifest, null, 2);
+  };
+
+  return (
+    <DialogWrapper
+      dialogType={DialogTypes.CreateFlow}
+      isOpen={!props.hidden}
+      subText={formatMessage(
+        'Your Teams adapter is configured for your published bot. Copy the manifest, open App Studio in Teams and add the manifest so you can test your bot in Teams'
+      )}
+      title={formatMessage('Teams Manifest')}
+      onDismiss={props.onDismiss}
+    >
+      <div>
+        <Text style={{ fontWeight: 700 }}>{formatMessage('Teams manifest for your bot:')}</Text>
+        <a
+          download={'teamsManifest.json'}
+          href={'data:text/plain;charset=utf-8,' + encodeURIComponent(generateTeamsManifest())}
+        >
+          <IconButton
+            ariaLabel={formatMessage('Download Icon')}
+            menuIconProps={{ iconName: 'Download' }}
+            styles={iconButtonStyle}
+          />
+        </a>
+        <IconButton
+          ariaLabel={formatMessage('Copy Icon')}
+          menuIconProps={{ iconName: 'Copy' }}
+          styles={iconButtonStyle}
+          onClick={() => {
+            copyCodeToClipboard();
+          }}
+        />
+      </div>
+      <TextField
+        multiline
+        componentRef={textFieldRef}
+        rows={25}
+        styles={{ root: { marginTop: '10px' }, fieldGroup: { backgroundColor: '#f3f2f1' } }}
+        value={generateTeamsManifest()}
+      />
+      <DialogFooter>
+        <DefaultButton
+          text={formatMessage('Close')}
+          onClick={() => {
+            props.onDismiss();
+          }}
+        />
+        <PrimaryButton
```

Comment:
why is this a PrimaryButton, this is a link

Reply:
The designs called for a button that opens teams. So we want button styling but link behavior.

Reply:
ok cool, can you put the link in a variable? what is that url for? do we need to aka.ms it?